### PR TITLE
Implement add project feature

### LINKED ISSUE #44: Implement add project feature

### PR DESCRIPTION
**Pull request by zrei**
Closes #44

### CODE REVIEW

File: src/test/java/arb/logic/parser/project/AddProjectCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        assertParseFailure(parser, VALID_TITLE_SKY_PAINTING + VALID_DEADLINE_SKY_PAINTING,
+                expectedMessage);
+    }
+
+    @Test
+    public void parse_invalidValue_failure() {
+        // invalid name
+        assertParseFailure(parser, INVALID_TITLE_DESC + DEADLINE_DESC_SKY_PAINTING, Title.MESSAGE_CONSTRAINTS);
+
+        // invalid deadline
```
Comment (zrei) [inline src/test/java/arb/logic/parser/project/AddProjectCommandParserTest.java:73]: Due to the validation for Deadlines not yet being implemented, this part of the test is commented out.

File: src/test/java/arb/logic/parser/project/AddProjectCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        //assertParseFailure(parser, TITLE_DESC_SKY_PAINTING + INVALID_DEADLINE_DESC, Deadline.MESSAGE_CONSTRAINTS);
```
Comment (zrei) [inline src/test/java/arb/logic/parser/project/AddProjectCommandParserTest.java:74]: Due to validation for Deadlines not yet being implemented, this part of the test for inalid values is commented out.

Comment (Johnchiahk) [reply src/test/java/arb/logic/parser/project/AddProjectCommandParserTest.java:74]: Okay!

File: src/test/java/arb/logic/commands/project/AddProjectCommandTest.java [hunk 3]
```diff
@@ -51,7 +52,8 @@ public void execute_duplicateProject_throwsCommandException() {
         ModelStub modelStub = new ModelStubWithProject(validProject);
 
         assertThrows(CommandException.class,
-                AddProjectCommand.MESSAGE_DUPLICATE_PROJECT, () -> addProjectCommand.execute(modelStub));
+                AddProjectCommand.MESSAGE_DUPLICATE_PROJECT, () ->
+                addProjectCommand.execute(modelStub, ListType.PROJECT));
```
Comment (Johnchiahk) [inline src/test/java/arb/logic/commands/project/AddProjectCommandTest.java:56]: Just to make sure, there shouldn't be any indentation on line 56?

Comment (Johnchiahk) [reply src/test/java/arb/logic/commands/project/AddProjectCommandTest.java:55]: Looks good since it is more readable!

**Review by Johnchiahk — APPROVED** [review]

**Pull request merged**